tests/unit_tests/drivers/test_jobscript_driver.py

```diff
@@ -62,72 +160,121 @@ def fixture_job_options(tmp_path, job_id, experiment_name, injected_input_files)
     return job_options
 
 
-@pytest.fixture(name="input_templates")
-def fixture_input_templates(tmp_path, job_options, parameters):
-    """Generate input templates."""
-
-    def templatify(list_of_keys, template_path):
-        """Create dict from list of keys."""
-        template_dict = {f: "{{ " + f + " }}" for f in list_of_keys}
-        # yaml format is used to ease read in of input file
-        template_path.write_text(yaml.safe_dump(template_dict))
-
-    # only add the second parameter
-    input_template_1 = tmp_path / "input_template_1.yaml"
-    templatify(list(job_options.to_dict().keys()) + parameters.names[-1:], input_template_1)
-
-    # add both parameters to the template
-    input_template_2 = tmp_path / "input_template_2.yaml"
-    templatify(parameters.names + ["input_1", "experiment_name"], input_template_2)
-
-    return input_template_1, input_template_2
-
-
-@pytest.fixture(name="injected_input_files")
-def fixture_injected_input_files(tmp_path, job_id, experiment_name):
-    """Fixture for the create input files."""
-    input_file_1 = tmp_path / str(job_id) / f"{experiment_name}_input_1_{job_id}.yaml"
-    input_file_2 = tmp_path / str(job_id) / f"{experiment_name}_input_2_{job_id}.yaml"
-    injected_input_files = {"input_1": input_file_1, "input_2": input_file_2}
-
-    return injected_input_files
-
-
 @pytest.fixture(name="jobscript_driver")
-def fixture_jobscript_driver(parameters, input_templates):
+def fixture_jobscript_driver(parameters, input_templates, executable):
     """Jobscript driver object."""
     input_template_1, input_template_2 = input_templates
 
     driver = JobscriptDriver(
         parameters=parameters,
         jobscript_template="",
-        executable="",
+        executable=executable,
         input_templates={"input_1": input_template_1, "input_2": input_template_2},
     )
 
     return driver
 
 
-def test_jobscript_driver_multiple_input_files(
-    jobscript_driver, job_options, injected_input_files, parameters
+@pytest.fixture(name="args_init")
+def fixture_args_init(
+    parameters,
+    jobscript_template,
+    executable,
+    input_template,
+    files_to_copy,
+    data_processor,
+    gradient_data_processor,
+    jobscript_file_name,
+    extra_options,
 ):
+    """Arguments to initialize a JobscriptDriver.
+
+    These arguments are meant for initialization with the default
+    constructor.
+    """
+    # pylint: disable=unused-argument
+    args_init = locals()
```

Comment:
We normally don't use the `locals` function. I understand the idea, but it makes everything super implicit. Also, this means you have to disable Pylint every time you use it. For cases like this, we create a fixture where the init arguments are stored in a dict. I know, it's not the most beautiful and pythonic way, but at least it's transparent.